sections/multirow.liquid

```diff
@@ -0,0 +1,437 @@
+{{ 'component-image-with-text.css' | asset_url | stylesheet_tag }}
+
+{%- style -%}
+  .section-{{ section.id }}-padding {
+    padding-top: {{ section.settings.padding_top | times: 0.75 | round: 0 }}px;
+    padding-bottom: {{ section.settings.padding_bottom | times: 0.75 | round: 0 }}px;
+  }
+
+  @media screen and (min-width: 750px) {
+    .section-{{ section.id }}-padding {
+      padding-top: {{ section.settings.padding_top }}px;
+      padding-bottom: {{ section.settings.padding_bottom }}px;
+    }
+  }
+{%- endstyle -%}
+
+{%- liquid
+  assign odd_class = 'image-with-text__grid--standard'
+  assign even_class = 'image-with-text__grid--reverse'
+  if section.settings.image_layout contains 'right'
+    assign odd_class = 'image-with-text__grid--reverse'
+    assign even_class = 'image-with-text__grid--standard'
+  endif
+
+  assign transparent_content = false
+  if section.settings.row_color_scheme == section.settings.section_color_scheme
+    assign transparent_content = true
+  endif
+
+  if transparent_content and settings.text_boxes_shadow_opacity == 0 and settings.text_boxes_border_thickness == 0 or settings.text_boxes_border_opacity == 0
+    assign padding_class = ' collapse-padding'
+  endif
+  if settings.text_boxes_border_thickness > 0 and settings.text_boxes_border_opacity > 0 and settings.media_border_thickness > 0 and settings.media_border_opacity > 0
+    assign borders_class = ' collapse-borders'
+  endif
+  unless transparent_content and settings.media_border_thickness > 0 and settings.text_boxes_shadow_opacity == 0 and settings.text_boxes_border_thickness == 0 or settings.text_boxes_border_opacity == 0
+    assign corners_class = ' collapse-corners'
+  endunless
+-%}
+
+<div class="multirow section-{{ section.id }}-padding gradient color-{{ section.settings.section_color_scheme }}">
+  <div class="page-width">
+    {%- for block in section.blocks -%}
+      <div class="image-with-text isolate{{ borders_class }}{{ corners_class }}{{ padding_class }}">
+        <div class="image-with-text__grid grid grid--gapless grid--1-col grid--{% if section.settings.desktop_image_width == 'medium' %}2-col-tablet{% else %}3-col-tablet{% endif %}{% if section.settings.image_layout contains 'alternate' %} {% cycle odd_class, even_class %}{% else %} {{ odd_class }}{% endif %}">
```

Comment:
+1 for other opinions, I'm curious how others feel.

Good idea to consider the rest of the codebase, but we could also use this as an opportunity to start pushing the codebase in a better direction, if we have broad agreement that more abstraction is better.

Reply:
I wondering if we need the `cycle` ? Could we leverage CSS selectors instead ? `:nth-child(even)` or `:nth-child(odd)`

Reply:
😅 fair. I didn't look too deeply into yet. I was just going through the PR description and comments 😬 Thanks for the context.

Reply:
@dan-menard I briefly looked at what abstracting more classes out of line 45 might look like. Abstracting all the global grid-related classes (a common example) ended up looking something like one of these options. Being able to keep the actual dom element's class attr cleaner could really be helpful at times, but I'd probably want to further consider how an approach like this would apply to other sections and types of classes before implementing. That said, any initial reactions from anyone? 
```
{%- capture grid_classes -%}
  grid grid--gapless grid--1-col grid--{% if section.settings.desktop_image_width == 'medium' %}2-col-tablet{% else %}3-col-tablet{% endif %}
{%- endcapture -%}
<div class="image-with-text__grid {{ grid_classes }}{% if section.settings.image_layout contains 'alternate' %}{% cycle odd_class, even_class %}{% else %}{{ odd_class }}{% endif %}">
```
```
{%- liquid
  assign grid_classes = 'grid grid--gapless grid--1-col'
  if section.settings.desktop_image_width == 'medium'
    assign grid_classes = grid_classes | append: 'grid--2-col-tablet'
  else
    assign grid_classes = grid_classes | append: 'grid--3-col-tablet'
  endif
-%}
<div class="image-with-text__grid {{ grid_classes }} {% if section.settings.image_layout contains 'alternate' %}{% cycle odd_class, even_class %}{% else %}{{ odd_class }}{% endif %}">
```